src/providers/RunSelectedLinesHoverProvider.ts

```diff
@@ -0,0 +1,42 @@
+import * as vscode from 'vscode';
+import { ICON_ID, RUN_SELECTION_COMMAND } from '../common';
+import { expandSelectionToFullLines } from '../util';
+
+/**
+ * Provides hover content for running selected lines in Deephaven.
+ */
+export const runSelectedLinesHoverProvider: vscode.HoverProvider = {
+  provideHover(_document, position, _token) {
+    const editor = vscode.window.activeTextEditor;
+
+    if (editor == null) {
+      return;
+    }
+
+    // Determine if hover is over a selected line
+    const isOverSelection = editor.selections.some(selection =>
+      expandSelectionToFullLines(editor.document)(selection).contains(position)
+    );
+
+    if (!isOverSelection) {
+      return;
+    }
+
+    // Tally total selected lines across all selections
+    const selectedLineCount = editor.selections.reduce(
+      (sum, selection) => sum + 1 + selection.end.line - selection.start.line,
+      0
+    );
+
+    const hoverContent = new vscode.MarkdownString(
+      `[$(${ICON_ID.runSelection}) Run Deephaven selected lines (${selectedLineCount})](command:${RUN_SELECTION_COMMAND})`,
```

Comment:
We should display the key binding in this as well if there is a binding set. I set mine to Ctrl+Shift+U for testing (couldn't find another one that was free really :laughing: )
Also I don't think showing the line count is really that helpful? I found it confusing at first, didn't understand what the number was for. Don't have a strong opinion on that though.

Reply:
I removed the line count, but I have not yet found any apis that expose the keybindings explicitly in a way that can be shown in the hover provider.